lib/writer/block-write-stream.js

```diff
@@ -95,6 +97,27 @@ class BlockWriteStream extends stream.Writable {
     this.open()
   }
 
+  /**
+   * @summary Determine whether an error is considered a
+   * transient occurrence, and the operation should be retried
+   * Errors considered potentially temporary are:
+   *   - Mac OS: ENXIO, EBUSY
+   *   - Windows: ENOENT, UNKNOWN
+   *   - Linux: EIO
+   * @private
+   * @param {Error} error - Error
+   * @returns {Boolean}
+   * @example
+   * BlockWriteStream.isTransientError(error)
+   */
+  static isTransientError (error) {
+    return error && (error.code === 'EIO' ||
+      error.code === 'ENXIO' ||
+      error.code === 'ENOENT' ||
+      error.code === 'EBUSY' ||
+      error.code === 'UNKNOWN')
```

Comment:
Would this be 'neater' if it used https://lodash.com/docs/4.17.4#find instead?

Reply:
I'd rather not have lodash anywhere near performance critical code, as our multipipage benchmarks revealed it was a source of allocations & slowness; a demonstration:

```
NANOBENCH version 2
> node error-code-check.js

# equals ⨉ 1000000
ok ~13 ms (0 s + 12832994 ns)

# branch equals ⨉ 1000000
ok ~23 ms (0 s + 23072749 ns)

# switch equals ⨉ 1000000
ok ~25 ms (0 s + 24974775 ns)

# branch / array includes ⨉ 1000000
ok ~35 ms (0 s + 35088306 ns)

# switch / array includes ⨉ 1000000
ok ~38 ms (0 s + 38391439 ns)

# lodash array includes ⨉ 1000000
ok ~180 ms (0 s + 180124552 ns)

# lodash branch array includes ⨉ 1000000
ok ~188 ms (0 s + 187953699 ns)

all benchmarks completed
ok ~502 ms (0 s + 502438514 ns)
```

Where the "branch equals" is the check we want here (platform dependant), and "equals" is what this initially is/was here.

Which makes use of lodash a slowdown of ~800% – even though minuscule at these time scales, the calls add up throughout and end up accounting for a noticeable slowdown.

Reply:
Would it make sense to globally disable the 'prefer lodash' linter settting? Or is it still okay to use lodash, just not in performance critical code?